jobfunnel/tests/test_tools.py

```diff
@@ -42,6 +44,148 @@
+    post_date_from_relative_post_age(job_list)
+    assert datetime.now().strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_yesterday_ago_pass(job_listings):
+    job_list = job_listings(attr_list[8:9])
+    post_date_from_relative_post_age(job_list)
+    yesterday = datetime.now() - timedelta(days=int(1))
+    assert yesterday.strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_1_day_ago_pass(job_listings):
+    job_list = job_listings(attr_list[9:10])
+    post_date_from_relative_post_age(job_list)
+    one_day_ago = datetime.now() - timedelta(days=int(1))
+    assert one_day_ago.strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_2_days_ago_pass(job_listings):
+    job_list = job_listings(attr_list[10:11])
+    post_date_from_relative_post_age(job_list)
+    two_days_ago = datetime.now() - timedelta(days=int(2))
+    assert two_days_ago.strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_1_month_ago_pass(job_listings):
+    job_list = job_listings(attr_list[11:12])
+    post_date_from_relative_post_age(job_list)
+    one_month_ago = datetime.now() - relativedelta(months=int(1))
+    assert one_month_ago.strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_2_months_ago_pass(job_listings):
+    job_list = job_listings(attr_list[12:13])
+    post_date_from_relative_post_age(job_list)
+    two_months_ago = datetime.now() - relativedelta(months=int(2))
+    assert two_months_ago.strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_1_year_ago_pass(job_listings):
+    job_list = job_listings(attr_list[13:14])
+    post_date_from_relative_post_age(job_list)
+    one_year_ago = datetime.now() - relativedelta(years=int(1))
+    assert one_year_ago.strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_2_years_ago_pass(job_listings):
+    job_list = job_listings(attr_list[14:15])
+    post_date_from_relative_post_age(job_list)
+    two_years_ago = datetime.now() - relativedelta(years=int(2))
+    assert two_years_ago.strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_1_epoch_ago_pass(job_listings):
+    job_list = job_listings(attr_list[15:16])
+    post_date_from_relative_post_age(job_list)
+    assert datetime(1970, 1, 1).strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_post_date_from_relative_ago_post_age_junk(job_listings):
+    job_list = job_listings(attr_list[16:17])
+    post_date_from_relative_post_age(job_list)
+    assert datetime(1970, 1, 1).strftime('%Y-%m-%d') == job_list[0]['date']
+
+
+def test_config_factory(configure_options):
+    config = config_factory(configure_options(
+        ['']), attr_list[17:18])[0]
+    assert config['some_option'] == 'option_value'
+
```

Comment:
excellent :)

One thing that would take your knowledge of pytest even further would be to write a parameterized test, this will let pytest know that it can run the same test method many times over with combinations of arguments. 

https://docs.pytest.org/en/latest/parametrize.html